modules/load_balancer/load_balancer.c

```diff
@@ -795,12 +807,28 @@ static void lb_update_max_loads(unsigned int ticks, void *param)
 				            dst->rmap[ri].resource->profile, &dst->profile_id);
 				old = dst->rmap[ri].max_load;
 
+				// if ( flags & LB_FLAGS_PERCENT_WITH_CPU ) { todo flags not avavilable here
 				/*
```

Comment:
Hi @liviuchircu - thanks for your feedback, I'm working on an improved PR on my local branch.

I'm a bit stuck on the documentation side - I've made some changes to the module README and was hoping to generate the xml file from this. My guesswork has me looking into the following command:

```
$ make modules=modules/load_balancer modules-readme
error: docbook.xsl not found (docbook-xsl)
make: *** [Makefile:240: tool-xsltproc] Error 1
```

As shown this isn't working for me, if this is still the way please can you help me to understand what is required! If I should just edit the XML then I'm happy to do that but I recall an automated process from a previous PR so wanted to check with an expert!

Reply:
Ah, sure thing!  In order to document the new modparam, edit `modules/load_balancer/docs/load_balancer_admin.xml`.  FYI, the `README` may be built as below, however there is no need to include it in your commit:

    make modules-readme modules=modules/load_balancer

Finally, you can actually build the HTML too in order to visualize your addition, just make sure you have the right packages (I have: `docbook-xml docbook-xsl docbook-xsl-doc-html html2text`), then just:

    make modules-docbook-html modules=modules/load_balancer
    chromium-browser modules/load_balancer/doc/load_balancer.html

Reply:
Excellent, thanks for that. It works exactly opposite to how I had remembered - the chrome testing was invaluable for checking the changes. I've used these instructions to add to the module documentation, reflecting a flag change in the recent commits as I've reached for a name which fits the approach.

Could you please take a look and see if there is anything I could make clearer in the docs or any further improvements to the patch.

The biggest weakness I have identified for this new approach is the interaction between the freeswitch and load_balancer modules. For my setup both are configured for 1s fidelity so every second we receive a heartbeat from FreeSWITCH and we also have a per second timer which retrieves those values and resets the "sessions since last heartbeat" counter. In an ideal configuration the heartbeat arrival would be the trigger for the counter reset. With my configuration values this issue is minimised however I am concerned that anyone using this with larger values for event_heartbeat_interval/fetch_freeswitch_stats might see some poor allocations, similarly if the user subscribes for heartbeats every 30s but fetches them every 5 second will see the counter resetting before the underlying data has been updated. I've attempted to include a description of this in the doc patch.

Similarly to the above, this new flag requires freeswitch stats to be enabled. Should I add a protection into the module to enforce this? I also considered whether it might be sensible to force matching values on event_heartbeat_interval/fetch_freeswitch_stats when this flag is set, to counter the point above.